R/parse_expr.R

```diff
@@ -116,17 +116,6 @@ parse_expr_assignment_lhs <- function(lhs, src, call) {
         "E1003", src, call)
     }
 
-    if (special == "compare") {
-      ## TODO: a good candidate for pointing at the source location of
-      ## the error.
-      odin_parse_error(
-        c("'compare()' expressions must use '~', not '<-'",
-          i = paste("Compare expressions do not represent assignents, but",
-                    "relationships, which we emphasise by using '~'.  This",
-                    "also keeps the syntax close to that for the prior",
-                    "specification in monty")),
-        "E1004", src, call)
-    }
```

Comment:
Hmm - compare(x) ~ Normal is handled fine, but what about if `compare(x) <- ` is written...

Reply:
That was an error before, so it will error now (I think), even if that error is now a bit mysterious, but that's ok

Reply:
I've updated the error text, which is a little less tidy, but hopefully less mysterious